kodilla-good-patterns/src/main/java/com/kodilla/good/patterns/challenges/FlighSearcher/FlightSearch.java

```diff
@@ -0,0 +1,46 @@
+package com.kodilla.good.patterns.challenges.FlighSearcher;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FlightSearch {
+
+    Gui gui = new Gui();
+    private FlightsBase flightsBase = new FlightsBase();
+    public List<Flight> searchDepartureCity(String departureCity)
+    {
+        System.out.println("Searching flights from: " + gui.getDepartureCity());
+        List<Flight> departures = flightsBase.getFlights().stream()
+                .filter(departure -> departure.getDepartureCity().equals(gui.getDepartureCity()))
+                .collect(Collectors.toList());
+        return departures;
+    }
```

Comment:
Problem jest z
Gui gui = new Gui();
Przekazujesz do metody parametr departureCity, z pozniej nie wykorzystujesz go. Zamiast tego wyciagasz 
gui.getDepartureCity() ktory jest null'em poniewaz Gui zostalo dopiero co stworzone (nowe, inne od tego ktore stworzyles w main)

Reply:
Poprawnie powinno to wygladac
    public List<Flight> searchDepartureCity(String departureCity)
    {
        System.out.println("Searching flights from: " + departureCity);
        List<Flight> departures = flightsBase.getFlights().stream()
                .filter(departure -> departure.getDepartureCity().equals(departureCity))
                .collect(Collectors.toList());
        return departures;
    }